### PR TITLE
Remove `nicolaslopezj:roles`

### PR DESCRIPTION
**Pull request by zarvox**
When trying to take the Meteor 2.3.4 update, we discovered that it was being held back by `nicolaslopezj:roles`, which has not been updated since 2018, which had a dependency on an older version of `accounts-base` than is available in Meteor 2.3.4.

Separately, we're interested in potentially making the `operator` role something that can have a finer granularity than "per server" (which is how it currently works).  This change series does some (but nowhere near all) of the refactoring work needed to support per-hunt operator status.

This is not meant to be a particularly pretty or DRY reimplementation of Roles.  This is meant to be a mostly-mechanical stopgap that unblocks taking updates.

### CODE REVIEW

File: imports/client/components/ProfilePage.tsx
Comment (zarvox) [inline imports/client/components/ProfilePage.tsx:57]: My best idea for how to make these not rely on the deprecated global calls is to add the route `/hunts/:huntId/hunters/:userId` for hunt-specific user lookup and then only present hunt-specific operator-y context on that page.

Then, the global user pages at `/users/:userId` could drop the operator-specific components.

File: imports/client/components/NotificationCenter.tsx
Comment (zarvox) [inline imports/client/components/NotificationCenter.tsx:373]: `pendingGuesses` might need to be restructured or made to work for non-operators or scoped to specific hunts or something to make this work reasonably in a tracking-operator-per-hunt world.

File: imports/server/hunts.ts
Comment (zarvox) [inline imports/server/hunts.ts:124]: Feel free to take a crack at winning a duel with the typechecker here; I gave up and moved on.

(In an ideal world, maybe we'd be able to synthesize these match patterns from the io-ts types or something?  Or use a `Match.Where` that calls the codec to see if it deserializes cleanly?  But maybe we'd rather find some other approach with smaller codesize?  🤷🏻‍♂️ )

File: imports/server/api/resources/users.ts
Comment (zarvox) [inline imports/server/api/resources/users.ts:40]: @grantaelliott as the maintainer of the only known jolly-roger API client: do you know if you actually rely on the presence/value of the `operator` field from the jolly-roger API for the POTW site or any of the other hunt sites you've written?  Would love to have the flexibility to drop this field or restructure it on a per-hunt basis so long as it wouldn't break you.

Comment (ebroder) [reply imports/server/api/resources/users.ts:40]: This is why we should switch to graphql :joy:

File: imports/server/hunts.ts
Comment (ebroder) [reply imports/server/hunts.ts:124]: It seems like the issue is that Typescript is inferring the type of `[String]` to be an array of unspecified size (`StringConstructor[]`), rather than a tuple type (`[StringConstructor]`), and that resolution happens at assignment time and isn't updated or refined when the object is passed into `check`.

File: imports/server/hunts.ts
Comment (ebroder) [inline imports/server/hunts.ts:89]:
```suggestion
  mailingLists: [String] as [StringConstructor],
```

File: imports/server/hunts.ts
Comment (ebroder) [inline imports/server/hunts.ts:108]:
```suggestion
    check(value, HuntShape);

    const huntId = Hunts.insert(value);
```

File: imports/server/hunts.ts
Comment (ebroder) [reply imports/server/hunts.ts:124]: Also we could use `Match.Where` but when I originally added the typing I didn't make the typing for that specific enough. Right now it's

```
        function Where(condition: (val: any) => boolean): Matcher<any>;
```

and I think it would need to be something like

```
        function Where<T>(condition: (val: any) => val is T): Matcher<T>;
```

**Comment by zarvox** [comment]
@ebroder Is this good now, or is there something else you'd like me to address first?

File: imports/client/components/ProfilePage.tsx
Comment (ebroder) [reply imports/client/components/ProfilePage.tsx:57]: Ugh. We don't need to solve this now but that seems messy. It's already pretty messy that there are 2 different ways to get to the profile list that look the same but show different things

File: imports/lib/models/base.ts
Comment (ebroder) [inline imports/lib/models/base.ts:43]: Hmm. This makes me wonder if we should figure out some way to make these rules off by default? We could have a Meteor method that enables them per-session or something like that and upgrade `loadFacades()` to, like, `enableAdminMode()` or whatever. Not a today problem but it makes me wonder if we can systematically prevent accidentally relying on these in the future.

**Review by ebroder — APPROVED** [review]

File: imports/lib/models/base.ts
Comment (zarvox) [reply imports/lib/models/base.ts:43]: That's a nice idea -- we could have these rules also check a global boolean or something that we can toggle via a Meteor method `breakGlass()`, and maybe have it automatically set a 10-minute timer to flip the boolean back to false?

Or it could just be a feature flag we can toggle from `/setup`

File: imports/client/components/ProfilePage.tsx
Comment (zarvox) [reply imports/client/components/ProfilePage.tsx:57]: Yeah -- I think we should probably nudge UI towards the hunt-scoped option.

I think if we're thinking about unrelated entities having accounts, it'd be pretty reasonable to restrict usage of `/users` and `/users/:userId` to admins and make everything else be hunt-scoped.  Might need to relocate `OwnProfilePage` to somewhere a little more generic.

File: imports/server/hunts.ts
Comment (ebroder) [reply imports/server/hunts.ts]: Delayed update: I believe that once DefinitelyTyped/DefinitelyTyped#55140 lands and we pick it up, we could replace this with `check(value, Match.Where(HuntCodec.is))`.

File: imports/server/api/resources/users.ts
Comment (grantaelliott) [reply imports/server/api/resources/users.ts:40]: POTW does not depend on the operator field.

**Pull request merged**
